themes/digital.gov/layouts/partials/core/featured-resource.html

```diff
@@ -1,66 +1,104 @@
+
+@example display a community in a shortcode:
+{{< featured-resource link = "/communities/communicators" kicker = "Join the Community" >}}
 
-@example Community shortcodes
-{{< featured-resource resourcePath = "/communities/communicators" kicker = "Join the Community" >}}
+@example allowed community resource paths: - communicators - multilingual -
+plain-language - social-media - user-experience - web-analytics-and-optimization
+- web-managers-forum */}}
+<!-- prettier-ignore-end -->
 
+{{ $resource := .resource_data }}
+{{ $title := "" }}
+{{ $summary := "" }}
+{{ $image := "" }}
+{{ $kicker := "" }}
+{{ $internalLink := "" }}
+{{ $externalLink := "" }}
+{{ $isExternal := false }}
 
-<!-- prettier-ignore -->
-Community resource paths:
-  - communicators
-  - multilingual
-  - plain-language
-  - social-media
-  - user-experience
-  - web-analytics-and-optimization
-  - web-managers-forum
-*/}}
+{{/* Check if page is an external link, otherwise return a hugo page object */}}
+{{ if or (strings.HasPrefix $resource.link "https://") (strings.HasPrefix $resource.link "http://") }}
+  {{ $isExternal = true }}
+  {{ $title = $resource.title }}
+  {{ $summary = $resource.summary }}
+  {{ $kicker = $resource.kicker }}
+  {{ $externalLink = $resource.link }}
+{{ else }}
+  {{ $page := .Site.GetPage $resource.link }}
+  {{ $title = $page.Title }}
+  {{ $internalLink = $page.Permalink }}
+  {{ $summary = $page.Params.summary }}
+  {{ $image = $page.Params.image }}
 
-{{- $resourcePath := .resourcePath -}}
-{{- $kicker := .kicker -}}
-{{- if $resourcePath -}}
-  {{- $resource := .Site.GetPage $resourcePath -}}
-  {{- if $resource -}}
+  {{/* Check if kicker is present as a custom key, otherwise use page parameter to set */}}
+  {{ if $resource.kicker }}
+    {{ $kicker = $resource.kicker }}
+  {{ else }}
+    {{ if $page.Params.kicker }}
+      {{ $kicker = $page.Params.kicker }}
+    {{ end }}
+  {{ end }}
+{{ end }}
 
-    <div class="dg-featured-resource">
-      <a
-        href="{{- $resource.Permalink | relURL -}}"
-        class="dg-featured-resource__body"
-      >
-        <div class="dg-featured-resource__text">
-          <p class="dg-featured-resource__text-kicker">
-            {{- if $kicker -}}
-              {{- $kicker | markdownify -}}
-            {{- else -}}
-              Featured Resource
-            {{- end -}}
-          </p>
-          <h2 class="dg-featured-resource__text-title">
-            {{- $resource.Params.title -}}
-          </h2>
-          <p class="dg-featured-resource__text-description">
-            {{- $resource.Params.summary -}}
-          </p>
+{{- if $resource -}}
+  <div class="dg-featured-resource">
+    <a
+      href="{{ if $isExternal }}
+        {{ $externalLink }}
+      {{ else }}
+        {{- $internalLink | relURL -}}
+      {{ end }}"
+      class="dg-featured-resource__body"
+    >
+      <div class="dg-featured-resource__text">
+        <p class="dg-featured-resource__text-kicker">
```

Comment:
While this does make it more readable; it also ties it too much to the text class.

Meaning, if we wanted to create a global kicker style, we couldn't do that with this component.

I'd recommend something like example below. That way it's a little more flexible in its placement.

```suggestion
        <p class="dg-featured-resource__kicker">
```

Reply:
Are Kicker styles really necessary in this project?